presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1011,6 +1018,86 @@ public void testTime(boolean insertWithPresto, ZoneId sessionZone)
         }
     }
 
+    /**
+     * Additional test supplementing {@link #testTime} with timestamp precision higher than expressible with {@code LocalTime}.
+     *
+     * @see #testTime
+     */
+    @Test
+    public void testTimeCoercion()
+            throws SQLException
+    {
+        testCreateTableAsAndInsertConsistency("TIME '00:00:00'", "TIME '00:00:00'");
+        testCreateTableAsAndInsertConsistency("TIME '00:00:00.000000'", "TIME '00:00:00.000000'");
+        testCreateTableAsAndInsertConsistency("TIME '00:00:00.123456'", "TIME '00:00:00.123456'");
+        testCreateTableAsAndInsertConsistency("TIME '12:34:56'", "TIME '12:34:56'");
+        testCreateTableAsAndInsertConsistency("TIME '12:34:56.123456'", "TIME '12:34:56.123456'");
+
+        // Cases which require using PgObject instead if PGobject with PostgreSQL JDBC driver
```

Comment:
"instead **of**". Also, are `PgObject` and `PGobject` different things? I couldn't find any class named `PgObject`

Reply:
fixed